### ISSUE TITLE
NFS client takes time to get up to speed

### ISSUE THREAD

**Issue by howardsamm** [zfsonlinux/zfs#3048]
I'm running an Oracle server connected to an HP DL380 based nas running SLES 11.3 and ZFS 6.3 with a pure ssd tank of mirrors. The two systems are connected back to back with 10Gb, and using DNFS are able to get sustained read speeds of over 900MB/s (peaking regularly at 1GB/s) (once zfsprefetch was disabled!!), the trouble is that it takes quite some time to get there. 
Throughput starts slow and builds. Is this something that can be tuned out? There is a ZIL on the NAS, but no L2ARC.

**Comment by behlendorf** [comment]
You can set the `zfs_prefetch_disable=1` module option to completely disable ZFSs prefetch engine.  It would be interesting to investigate why the prefetch engine is internally fetching the wrong blocks.  Or perhaps those blocks just end up rolling out of the cache before they're needed and have to be read again.

**Comment by howardsamm** [comment]
I found that performance on the nas _looked_ great, but it wasn't translating server side.
zpool iostat would show around 3-400MB/s (data set is lz4 compressed), when I disabled prefetch it dropped to 100-150MB/s but the server saw a huge increase in usable data. It's definitely worth noting!
The slow starting seems to be getting better, I was wondering if it was an Oracle or NFS issue, but couldn't find any other reports of similar issues on the net.. 
By the way, zfs on linux is great!! I've been using Openinidana for a while now, but the performance under Linux is better once tuned (with SSD pools, the queue/nr_requests on the disks makes a big difference depending on workload).
